source/adapters/hip/command_buffer.hpp

```diff
@@ -9,7 +9,222 @@
 //===----------------------------------------------------------------------===//
 
 #include <ur/ur.hpp>
+#include <ur_api.h>
 
-/// Stub implementation of command-buffers for HIP
+#include "context.hpp"
+#include <hip/hip_runtime.h>
+#include <memory>
 
-struct ur_exp_command_buffer_handle_t_ {};
+static inline const char *getUrResultString(ur_result_t Result) {
```

Comment:
A bit unrelated to this PR but since this is the second time we are replicating this from the L0 backend we should maybe consider implementing this in a common location for all backends to use.

Reply:
Expanding on this, I think if we're going to do this it'd be better done as a separate PR since introducing common code to UR adapters is quite a bit outside the scope of this task and our feature in general, so resolving this comment.